app/main/views/frameworks.py

```diff
@@ -329,31 +330,28 @@ def framework_supplier_declaration(framework_slug, section_id=None):
                     current_user.email_address
                 )
 
-                next_section = content.get_next_editable_section_id(section_id)
-                if next_section:
-                    return redirect(
-                        url_for('.framework_supplier_declaration',
-                                framework_slug=framework['slug'],
-                                section_id=next_section))
+                if next_section and not request.form.get('save_and_return_to_overview', False):
+                    # Go to the next section.
+                    return redirect(url_for(
+                        '.framework_supplier_declaration',
+                        framework_slug=framework['slug'],
+                        section_id=next_section.id
+                    ))
                 else:
-                    url = "{}/declaration_complete".format(
-                        url_for('.framework_dashboard',
-                                framework_slug=framework['slug']))
-                    flash(url, 'declaration_complete')
-                    return redirect(
-                        url_for('.framework_dashboard',
-                                framework_slug=framework['slug']))
+                    # Otherwise they have reached the last page of their declaration.
+                    # Return to the overview.
+                    return redirect(url_for('.framework_supplier_declaration_overview', framework_slug=framework_slug))
             except APIError as e:
                 abort(e.status_code)
 
     return render_template(
         "frameworks/edit_declaration_section.html",
         framework=framework,
+        next_section=next_section,
         section=section,
         declaration_answers=all_answers,
-        is_last_page=is_last_page,
         get_question=content.get_question,
-        errors=errors,
+        errors=errors
```

Comment:
I think they harm readability. For instance. How long does it take to work out what this does?
`x = lambda: 'foo',`
any small gain like reordering arguments and/ or prettier diffs is negated by the inconsistency presented by statements like:
```
x= 'foo', 
len(x)
```
In python commas can _do_ something. For instance one is a string, one is a tuple:
```
('foo')
('foo',)
```
Unless we're asking the comma to _do_ something then it being there has the potential to mislead. 

Why is it there if it's not doing anything? Would be my argument in a nutshell.

We also don't write:
`x=[foo, bar, baz,]`